core/contracts/Registry.sol

```diff
@@ -85,38 +85,38 @@ contract Registry is RegistryInterface, MultiRole {
         }
 
         address[] memory partiesForEvent = parties;
-        emit NewDerivativeRegistered(derivativeAddress, partiesForEvent);
+        emit NewDerivativeRegistered(derivativeAddress, msg.sender, partiesForEvent);
     }
 
     function isDerivativeRegistered(address derivative) external view returns (bool isRegistered) {
         return derivativePointers[derivative].valid == PointerValidity.Valid;
     }
 
-    function getRegisteredDerivatives(address party) external view returns (RegisteredDerivative[] memory derivatives) {
+    function getRegisteredDerivatives(address party) external view returns (address[] memory derivatives) {
```

Comment:
Does the dapp use this function? If so, does its call need to be updated?

Reply:
Yes, it absolutely does. PTAL.